### PR TITLE
feat:  E2EI-certificate renewal logic

### PR DESCRIPTION
**Pull request by arjita-mitra**
## Description
This PR updates the E2EI certificate renewal process in Wire, introducing automated renewal based on the Identity Provider's configuration and user's certificate status.

### Key Changes:
**Automated Renewal Logic:** Certificates are renewed automatically if conditions set by the IdP are met, enhancing user experience by minimizing manual intervention.

**User-Action Based Renewal:** In cases where user action is required, notifications guide users through the renewal process within the Wire app.

**Renewal Timing Calculation:** Renewal prompts are based on a calculated schedule considering certificate validity, message storage duration, grace period, and a randomized time factor to stagger renewals.

These changes streamline the E2EI renewal process, ensuring secure and uninterrupted usage of the Wire app.

## Screenshots/Screencast (for UI changes)
Screenshot 2023-12-21 at 10 49 25

## Checklist

- [x] PR has been self reviewed by the author;
- [x] Hard-to-understand areas of the code have been commented;
- [x] If it is a core feature, unit tests have been added;

### CODE REVIEW

File: .prettierignore [hunk 1]
```diff
@@ -10,3 +10,4 @@ CHANGELOG.md
 node_modules
 npm-debug.log
 yarn-error.log
+*.yaml
```
Comment (atomrc) [inline .prettierignore]: why is this change here? 🤔

Comment (arjita-mitra) [reply .prettierignore]: I am not sure, preetier added it I assume.

File: bin/push_docker.js [hunk 2]
```diff
@@ -54,18 +54,18 @@ tags.push(`${repository}:${versionTag}`);
 
 var configurationEntry;
 if (versionTag.includes('production')) {
-  configurationEntry = "wire-web-config-default-master";
+  configurationEntry = 'wire-web-config-default-master';
```
Comment (przemvs) [inline bin/push_docker.js:57]: Hmm.. I think it also comes from prettier ? All changes in this file

File: .prettierignore (hunk 1 above)
Comment (atomrc) [reply .prettierignore]: it's not adding it automatically

File: src/script/E2EIdentity/E2EIdentityEnrollment.ts [hunk 3]
```diff
@@ -54,7 +61,11 @@ type EnrollmentConfig = {
   discoveryUrl: string;
   gracePeriodInMs: number;
 };
+
+const gracePeriodMS = 7 * TimeInMillis.DAY; //GP
```
Comment (atomrc) [inline src/script/E2EIdentity/E2EIdentityEnrollment.ts]: I'm a little confused what is this graceperiod compared to the grace period that is given in the `initialize` function 🤔 . 

Maybe the naming needs to be improved there

File: bin/push_docker.js (hunk 2 above)
Comment (arjita-mitra) [reply bin/push_docker.js:57]: yaa, i haven't touched this file.

Comment (atomrc) [reply bin/push_docker.js:57]: most likely, yes

File: .prettierignore (hunk 1 above)
Comment (arjita-mitra) [reply .prettierignore]: I haven't added it, I did ran yalc though to get core changes. Not sure what caused this.

Comment (atomrc) [reply .prettierignore]: let's remove it from this PR then :)

File: src/script/E2EIdentity/E2EIdentityEnrollment.test.ts [hunk 4]
```diff
@@ -162,4 +191,122 @@ describe('E2EIHandler', () => {
       }),
     );
   });
+
+  it('should call renewCertificate when conditions are met', async () => {
```
Comment (atomrc) [inline src/script/E2EIdentity/E2EIdentityEnrollment.test.ts:195]: nice coverage 🫶

File: src/script/E2EIdentity/E2EIdentityEnrollment.ts [hunk 5]
```diff
@@ -92,32 +103,140 @@ export class E2EIHandler extends TypedEventEmitter<Events> {
...
-  public async enroll(refreshActiveCertificate: boolean = false) {
+  /**
+   * Used to clean the state/storage after a failed run
+   */
+  private async cleanUp() {
+    // Remove the url parameters of the failed enrolment
+    removeUrlParameters();
+    // Clear the oidc service progress
+    this.oidcService = new OIDCService();
+    await this.oidcService.clearProgress();
```
Comment (atomrc) [inline src/script/E2EIdentity/E2EIdentityEnrollment.ts:219]: should those 2 lines be in reversed order? It feels we are clearing the progress of a completely new instance

File: src/script/E2EIdentity/E2EIdentityEnrollment.ts [hunk 6]
```diff
@@ -170,11 +291,14 @@ export class E2EIHandler extends TypedEventEmitter<Events> {
 
       this.currentStep = E2EIHandlerStep.SUCCESS;
       this.showSuccessMessage();
-      // Remove the url parameters after enrolment
-      removeUrlParameters();
-      this.emit('enrollmentSuccessful');
```
Comment (atomrc) [inline src/script/E2EIdentity/E2EIdentityEnrollment.ts:175]: Please do not remove this event, we need it

**Review by atomrc — CHANGES_REQUESTED** [review]

File: src/script/E2EIdentity/E2EIdentityEnrollment.ts (hunk 3 above)
Comment (arjita-mitra) [reply src/script/E2EIdentity/E2EIdentityEnrollment.ts]: I will be using the same gracePeriod thats coming from TM so this will be removed.

File: src/script/E2EIdentity/E2EIdentityEnrollment.ts (hunk 5 above)
Comment (aweiss-dev) [reply src/script/E2EIdentity/E2EIdentityEnrollment.ts:219]: The progress is stored in localStorage, not depending on the instance. We just make sure that we have a oidcService here and can call clearProgress.

**Review by atomrc — APPROVED** [review]
